firefox-ios/Client/Frontend/TrackingProtection/BlockedTrackersTableController.swift

```diff
@@ -0,0 +1,385 @@
+        self.dismiss(animated: true)
+    }
+
+    // MARK: - UITableViewDataSource
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return model.getItems().count
+    }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        if section == 0 {
+            let headerView = tableView.dequeueReusableHeaderFooterView(
+                withIdentifier: BlockedTrackersHeaderView.cellIdentifier
+            ) as! BlockedTrackersHeaderView
+            let totalTrackerBlocked = String(model.contentBlockerStats?.total ?? 0)
+            let trackersText = String(format: .Menu.EnhancedTrackingProtection.trackersBlockedLabel,
+                                      totalTrackerBlocked)
+            headerView.totalTrackersBlockedLabel.text = trackersText
+            headerView.applyTheme(theme: currentTheme())
+            return headerView
+        }
+        return nil
+    }
+
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return UITableView.automaticDimension
+    }
+
+    // MARK: Notifications
+    func handleNotifications(_ notification: Notification) {
+        switch notification.name {
+        case .DynamicFontChanged:
+            adjustLayout()
+        default: break
+        }
+    }
+
+    // MARK: View Transitions
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        adjustLayout()
+    }
+
+    override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransition(to: size, with: coordinator)
+        coordinator.animate(alongsideTransition: { _ in
+            self.adjustLayout()
+        }, completion: nil)
+    }
+
+    func adjustLayout() {
+        for cell in blockedTrackersTableView.visibleCells {
+            if let blockedTrackerCell = cell as? BlockedTrackerCell {
+                blockedTrackerCell.invalidateIntrinsicContentSize()
+                blockedTrackerCell.setNeedsLayout()
+            }
+        }
+    }
+
+    // MARK: - Themable
+    private func currentTheme() -> Theme {
+        return themeManager.getCurrentTheme(for: windowUUID)
+    }
+
+    func applyTheme() {
+        let theme = currentTheme()
+        overrideUserInterfaceStyle = theme.type.getInterfaceStyle()
+        view.backgroundColor = theme.colors.layer1
+        blockedTrackersTableView.backgroundColor = theme.colors.layer1
+        blockedTrackersTableView.layer.borderColor = theme.colors.borderPrimary.cgColor
+        blockedTrackersTableView.reloadData()
```

Comment:
why are we reloading data here? haven't seen this being done in applyTheme, so want to make sure we have a good use case.